src/modules/olcs/Service.js

```diff
@@ -24,6 +67,96 @@ const Service = class {
   getManager() {
     return this.manager_;
   }
+
+  /**
+   * @export
+   * @return {number} the min tilt.
+   */
+  getMinTilt() {
+    return 0;
+  }
+
+  /**
+   * @export
+   * Almost Pi / 2
+   * @return {number} the max tilt.
+   */
+  getMaxTilt() {
+    return 7 * Math.PI / 16;
+  }
+
+  /**
+   * @private
+   */
+  cameraFromPermalink_() {
+    const manager = this.manager_;
+    const stateManager = this.ngeoStateManager_;
+    const scene = manager.getOl3d().getCesiumScene();
+    const camera = scene.camera;
+
+    const lon = stateManager.getInitialNumberValue('3d_lon');
+    const lat = stateManager.getInitialNumberValue('3d_lat');
+    const elevation = stateManager.getInitialNumberValue('3d_elevation');
+    const destination = Cesium.Cartesian3.fromDegrees(lon, lat, elevation);
+
+    const heading = Cesium.Math.toRadians(
+      stateManager.getInitialNumberValue('3d_heading') || 0);
+    const pitch = Cesium.Math.toRadians(
+      stateManager.getInitialNumberValue('3d_pitch') || 0);
+    const roll = 0;
+    const orientation = {heading, pitch, roll};
+
+    camera.setView({
+      destination,
+      orientation
+    });
+  }
+
+  /**
+   * @private
+   */
+  setupPermalink_() {
+    const manager = this.manager_;
+    const scene = manager.getOl3d().getCesiumScene();
+    const camera = scene.camera;
+
+    camera.moveEnd.addEventListener(this.ngeoDebounce_(() => {
+      const position = camera.positionCartographic;
+      this.ngeoStateManager_.updateState({
+        '3d_enabled': true,
+        '3d_lon': Cesium.Math.toDegrees(position.longitude).toFixed(5),
+        '3d_lat': Cesium.Math.toDegrees(position.latitude).toFixed(5),
+        '3d_elevation': position.height.toFixed(0),
+        '3d_heading': Cesium.Math.toDegrees(camera.heading).toFixed(3),
+        '3d_pitch': Cesium.Math.toDegrees(camera.pitch).toFixed(3)
+      });
+    }, 1000, true));
+
+    this.$rootScope.$watch(
+      () => this.manager_.getOl3d().getEnabled(),
+      this.on3dToggle_.bind(this)
+    );
+  }
+
+  /**
+   * @private
+   */
+  teardownPermalink_() {
```

Comment:
I would expect the 3d state to be removed from the state manager and that the state manager would then automatically take care of the permalink.

So better name it `remove3DState` and change the code here.